_posts/2022-05-31-introduction-to-project-panama-part-1.markdown

```diff
@@ -216,35 +226,33 @@ public static void main(String[] args) throws Throwable {
 }
 ```
 
-Note that we are allowed to cast a return value to _int_, because of the value layout of a return value defined within _printf_ **FunctionDescriptor**.
-
-In a real-life, instead of _int_ as a return value there possibly would be just a **MemorySegment** that you'd need to read
-from to get a value of a type (a struct, for instance). I'll show you more examples in the following parts!
+Note that we are allowed to cast a return type to _int_, because a return value layout carrier type of C _printf_ is _int.class_.
 
-### Task: get things together!
+### 5. Get things together!
 
-Let's summarize a few key aspects of C programs written in Java:
-* it is necessary to outline that you need to start a memory session only when you are actually about to allocate memory segments,
-  so, an entry point to the program should start with a session declaration;
-* Use a session within the try-with-resource block to make sure that a session is closed once you're done;
-* It's up to use to decide which API to use to allocate memory segments - through an allocator or a session directly. Although, if you dig into the implementation you'll notice the **SegmentAllocator** has a couple of session safety checks.
-* Linker, symbol lookup objects, value/memory layouts, and method handles are the static object.
+To summarise, invoking a C function from Java:
+* Memory session (**MemorySession**) or a segment allocator (**SegmentAllocator**) are key APIs to perform memory allocation;
+* Memory session should be declared with a try-with-resources to be automatically closed;
+* There are several options to allocate memory segments - through an allocator or a session directly;
+* Linker, symbol lookup objects, value/memory layouts, and method handles are static objects.
 
 
 The full code listing can be found below!
```

Comment:
The full code can be found [link to repo] (I guess you don't actually the full code in the article itself)